src/idl_gen_cpp.cpp

```diff
@@ -1783,6 +1937,9 @@ class CppGenerator : public BaseGenerator {
     }
     GenOperatorNewDelete(struct_def);
     GenDefaultConstructor(struct_def);
+    if (opts_.g_cpp_std >= cpp::CPP_STD_11) {
```

Comment:
Maybe move `(opts_.g_cpp_std >= cpp::CPP_STD_11)`  to `GenCpp11CopyMoveCtorAndAssigOpDecls` (`GenCopyMoveCtorAndAssigOpDecls`)?

Reply:
Sure, done.